src/freenet/pluginmanager/PluginManager.java

```diff
@@ -942,6 +952,27 @@ public FredPluginFCP getFCPPlugin(String plugname) {
 		return null;
 	}
 
+    /**
+     * Get the {@link FredPluginFCPMessageHandler.ServerSideFCPMessageHandler} of the plugin with
+     * the given class name.
```

Comment:
IMHO the fact that the plugin and the message handler are the same object is irrelevant to the outside calling code: The FCPPluginClient for example only cares about the fact that the ServerSideFCPMessageHandler IS a ServerSideFCPMessageHandler and that it thus can deliver messages to it, it doesn't need any other functionality of it. Thus it doesn't need to know that it is in fact more than just that.

Not promising more than the one interface is a good idea for future code quality IMHO: It will allow us to change the architecture of the way plugins are loaded to allow the plugin to specify DIFFERENT objects to implement the dozens of interfaces which there are. This will cut down the size of their main plugin class, which is indicated as a good idea in the main class of WOT already because it is HUGE: https://github.com/freenet/plugin-WebOfTrust/blob/50de01d423694abd1e8f71fa5563b304385c6d6a/src/plugins/WebOfTrust/WebOfTrust.java
All its handlers of plugin API, such as the FCP message handlers, are already redirected by wrapper functions to other classes anyway, which could be avoided by allowing the interfaces to be implemented by something else. Cut the problem at its root.
The uglyness is best visible for example at this function:
https://github.com/freenet/plugin-WebOfTrust/blob/50de01d423694abd1e8f71fa5563b304385c6d6a/src/plugins/WebOfTrust/WebOfTrust.java#L3770
Nobody would expect a class WebOfTrust to implement a "getString()".

ACK to keep as is?